### PR TITLE
Changes in the correlator init, GEVP and docstrings & tag format includes dict and prange

### PR DESCRIPTION
**Pull request by JanNeuendorf**
Okay, i repulled and redid the changes in the correlators file only. 
Are you sure, that i should not update my example data?

### CODE REVIEW

**Comment by fjosw** [comment]
The only thing I did is adjusting the tags to the new format, i.e. removing the excessive `0|0|`. Did you make any other changes? Or in other words. Does the example still work properly?

**Comment by fjosw** [comment]
Now the tests are failing because you simply overwrote my patch 90717ba85d6d9715dd4aae016e17da6995e08906 please include these changes in your version for the tests to pass.

**Comment by JanNeuendorf** [comment]
Yes, i just tested it, the examples still run on my machine. If this ever breaks because of an update to the format, i can just recreate them

**Comment by fjosw** [comment]
Thanks! Just a small suggestion: Do you see any problem in merging the two parameters `return_list` and `sorting`? It appears to me as if just having one parameter could give the same functionality.

**Comment by JanNeuendorf** [comment]
Okay, so it would be 
return_list=none to return a single value 
and return_list="Eigenvalue" for the sorting? 
sounds a bit confusing, but it would work. 
How about you allow the merge and i will change it for the next commit?

**Comment by JanNeuendorf** [comment]
Okay, i named the argument sorted_list 
it is None by default, so it does not return a list and you can set it to a string to give the sorting method.

**Comment by JanNeuendorf** [comment]
Okay, i changed the get_Corr_from_dic() and write_Corr_to_dict() to replace the old list "tag" with a dictionary. The dictionary has a key that is also named "tag", which holds the list, which is replaced. 
This sounds confusing, but @s-kuberski  explained it nicely. The dump method will always use the dictionary and include the prange in it, if available. The load method checks, if the tag is a list or a dic and proceeds accordingly. We can decide later, if we want to break this compatibility for simplicity. 
I think, this is actually a good solution, because we could include any optional parameters Corr could get in the future in this dictionary.

**Comment by fjosw** [comment]
Looks good, maybe @s-kuberski can also have a look before I merge the PR.
Closes #46

File: pyerrors/input/json.py [hunk 1]
```diff
@@ -395,7 +400,17 @@ def get_Array_from_dict(o):
         return np.reshape(ret, layout)
 
     def get_Corr_from_dict(o):
-        taglist = o.get('tag')
+        if isinstance(o.get('tag'), list):  # supports the old way
+            taglist = o.get('tag')  # This had to be modified to get the taglist from the dictionary
+            temp_prange = None
+        else:
```
Comment (fjosw) [inline pyerrors/input/json.py:406]: For safety reasons I would suggest to check whether `o.get('tag')` is a `dict` and raise an Exception if the tag is neither `list` nor `dict`.

**Comment by fjosw** [comment]
I would also advocate to add an additional test to pyerrors/tests/io_test.py in which it is checked whether the new dictionary tag and specifically the `prange` is correctly reconstructed from a `json` file.

File: pyerrors/input/json.py (hunk 1 above)
Comment (JanNeuendorf) [reply pyerrors/input/json.py]: Yes, i added the exception

Comment (s-kuberski) [reply pyerrors/input/json.py]: I agree. Too bad, that we already have to care about backwards compatibility, but I hope, we don't have to include too many of such cases. The tag-dictionary will certainly help in this regard.

**Comment by JanNeuendorf** [comment]
I agree that tests are needed. But maybe we should all get on one version and **then** decide how to test it

File: pyerrors/input/json.py (hunk 1 above)
Comment (JanNeuendorf) [reply pyerrors/input/json.py]: I guess as long as it is pre 2.0, Fabian could decide to break the format. 
One could easily update data into the new format.

**Comment by s-kuberski** [comment]
Looks good to me. In principle, we are now in the position to remove the leading and trailing None's from the JSON file using a 'padding' key in the dict, but this is not too important.

**Comment by fjosw** [comment]
For the future I would suggest that all new features have to come with basic tests. In this case the additional functionality can be easily tested by adding a line or two, I can do that on the fly.

**Pull request merged**